pandas/core/arrays/period.py

```diff
@@ -190,7 +190,9 @@ def _from_sequence(
             copy: bool = False,
     ) -> ABCPeriodArray:
         if dtype:
-            freq = dtype.freq
+            freq = dtype.freq  # type: ignore
```

Comment:
This is the line throwing `error: "PeriodDtype" has no attribute "freq"`. `PeriodDtype`'s `__new__` method uses a dictionary to cache frequencies, so it's more complex than the `self.freq = ...` machinery I think mypy expects.

Reply:
Would adding `freq = None  # type: XXX` as a class variable resolve this?

Reply:
Adding `freq = None` to the `PeriodDtype` class without any annotation at all did the trick.